src/mysql.c

```diff
@@ -270,12 +270,16 @@ static MYSQL *getconnection(mysql_database_t *db) {
   }
   db->is_connected = 0;
 
+  /* Close the old connection before initializing a new one. */
+  if (db->con != NULL) {
+    mysql_close(db->con);
+    db->con = NULL;
+  }
+
+  db->con = mysql_init(NULL);
```

Comment:
You're turning this into an unconditional `mysql_init`, which is probably not what you intended. Maybe the following would be more efficient:
```c
  /* Close the old connection before initializing a new one. */
  if (db->con != NULL) {
    mysql_close(db->con);
  } else {
    db->con = mysql_init(NULL);
    if (db->con == NULL) {
      ERROR("mysql plugin: mysql_init failed: %s", mysql_error(db->con));
      return NULL;
    }
  }
```
? Or even (to minimize changes):
```c
  if (db->con == NULL) {
    db->con = mysql_init(NULL);
    if (db->con == NULL) {
      ERROR("mysql plugin: mysql_init failed: %s", mysql_error(db->con));
      return NULL;
    }
  } else {
    /* Close the existing connection before reconnecting. */
    mysql_close(db->con);
  }
```

Reply:
Never mind, what I said was silly. Should have read the docs better. `mysql_close` deallocates the connection object, so it needs to be initialized again. Your proposed code does it correctly, mine doesn't.

Reply:
Yep, the return on line 266 ensures that if we get this far the connection has dropped regardless and should be cleaned up.